# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1236319624 18000
#      Fri Mar 06 01:07:04 2009 -0500
# Node ID 485eabc0cfece1d0b97ce7d21c72ca9a3e7de62a
# Parent  354179c24c79dcd00c2d14ca318661f0ba0d0929
__go_draw_axes__.m: Preserve the order of axes' children when pruning the handles for the {x,y,z}labels and title.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,15 @@
+2009-03-06  Ben Abbott <bpabbott@mac.com>
+
+	* plot/__go_draw_axes__.m: Preserve the order of axes' children
+	when pruning the handles for the {x,y,z}labels and title.
+	* plot/legend.m: Change demo to add colors to lines. This change
+	reveals the problem (above) with the ordering of the axes' children.
+
 2009-03-05  Jaroslav Hajek  <highegg@gmail.com>
 
 	* linear-algebra/condest.m: Use lu rather than splu. Use relative tols
 	for tests.
 
 2009-03-05  John W. Eaton  <jwe@octave.org>
 
 	* deprecated/spdiag.m: Ensure sparse result.
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -289,18 +289,21 @@ function __go_draw_axes__ (h, plot_strea
     zautoscale = strcmpi (axis_obj.zlimmode, "auto");
     cautoscale = strcmpi (axis_obj.climmode, "auto");
     cdatadirect = false;
     truecolor = false;
 
     fputs (plot_stream, "set clip two;\n");
 
     kids = axis_obj.children;
-    kids = setdiff (kids, [axis_obj.xlabel; axis_obj.ylabel; ...
-                           axis_obj.zlabel; axis_obj.title]);
+    ## Remove the axis labels and title from the children, and
+    ## preserved the original order.
+    [jnk, k] = setdiff (kids, [axis_obj.xlabel; axis_obj.ylabel; ...
+                               axis_obj.zlabel; axis_obj.title]);
+    kids = kids (sort (k));
 
     if (nd == 3)
       fputs (plot_stream, "set parametric;\n");
       fputs (plot_stream, "set style data lines;\n");
       fputs (plot_stream, "set surface;\n");
       fputs (plot_stream, "unset contour;\n");
     endif
 
diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -238,15 +238,16 @@ endfunction
 %! plot(1:10, 1:10);
 %! title("a very long label can sometimes cause problems");
 %! legend({"hello world"}, "location", "northeastoutside")
 
 %!demo
 %! clf
 %! labels = {};
 %! for i = 1:5
-%!     plot(1:100, i + rand(100,1)); hold on;
+%!     h = plot(1:100, i + rand(100,1)); hold on;
+%!     set (h, "color", get (gca, "colororder")(i,:))
 %!     labels = {labels{:}, cstrcat("Signal ", num2str(i))};
 %! endfor; hold off;
 %! title("Signals with random offset and uniform noise")
 %! xlabel("Sample Nr [k]"); ylabel("Amplitude [V]");
 %! legend(labels, "location", "southoutside")
 %! legend("boxon")
